src/mbgl/gfx/offscreen_texture.hpp

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+#include <mbgl/gfx/renderable.hpp>
+#include <mbgl/util/image.hpp>
+
+namespace mbgl {
+namespace gfx {
+
+class Texture;
+
+class OffscreenTexture : public gfx::Renderable {
+protected:
+    OffscreenTexture(const Size size_, std::unique_ptr<RenderableResource>&& resource_)
+        : gfx::Renderable(size_, std::move(resource_)) {
+    }
+
+public:
+    virtual ~OffscreenTexture() = default;
+
+    virtual PremultipliedImage readStillImage() = 0;
```

Comment:
nit: const method?

Reply:
Nope, reading the image from GL requires modifying internal state.